satellite/overlay/selectednodescache.go

```diff
@@ -0,0 +1,161 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package overlay
```

Comment:
Let's move this to a separate package `satellite/overlay/nodeselection/cache.go`, node selection will end up having quite a bit more logic in it.